### PR TITLE
Android fcm (gcm under the hood)

### PR DESCRIPTION
**Pull request by v-fekalu**

### CODE REVIEW

File: Android/notification-hubs-sdk-e2etestapp/src/main/java/com/microsoft/windowsazure/messaging/e2etestapp/ApplicationContext.java [hunk 1]
```diff
@@ -81,4 +84,12 @@ public static String getLogPostURL() {
 	public static String getGCMSenderId() {
 		return PreferenceManager.getDefaultSharedPreferences(mContext).getString(Constants.PREFERENCE_GCM_SENDER_ID, "");
 	}
+
+	public static String getFCMSenderId() {
+		return PreferenceManager.getDefaultSharedPreferences(mContext).getString(Constants.PREFERENCE_FCM_SENDER_ID, "");
+	}
+
+	public static Boolean getUseGcm(){
+		return PreferenceManager.getDefaultSharedPreferences(mContext).getBoolean(Constants.PREFERENCE_USE_DEPRECATED_GCM, false);
```
Comment (MatkovIvan) [inline Android/notification-hubs-sdk-e2etestapp/src/main/java/com/microsoft/windowsazure/messaging/e2etestapp/ApplicationContext.java]: ~I guess storing the type is better than just bool~
Let's just check if the id exists

File: Android/notification-hubs-sdk/build.gradle [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
-def version = 0.5
+def version = 0.6
 def PUBLISH_ARTIFACT_ID = 'notification-hubs-android-sdk'
 def groupId = 'com.microsoft.azure'
 
 android {
     compileSdkVersion 21
-    buildToolsVersion "21.1.2"
+    buildToolsVersion '21.1.2'
```
Comment (MatkovIvan) [inline Android/notification-hubs-sdk/build.gradle]: It's time to update it

File: Android/notification-hubs-sdk-e2etestapp/src/main/java/com/microsoft/windowsazure/messaging/e2etestapp/ApplicationContext.java (hunk 1 above)
Comment (v-fekalu) [reply Android/notification-hubs-sdk-e2etestapp/src/main/java/com/microsoft/windowsazure/messaging/e2etestapp/ApplicationContext.java]: SwitchPreference saves true/false if we will not do additional handling - in that case we need in getBoolean. When value is missing the default value "false" is passed as second argument to function getBoolean(..., false).

File: Android/notification-hubs-sdk/build.gradle (hunk 2 above)
Comment (v-fekalu) [reply Android/notification-hubs-sdk/build.gradle]: Updated

**Pull request merged**